### ISSUE TITLE
build error: dump struct `&&char`field fail

### ISSUE THREAD

**Issue by kahsa** [vlang/v#16008]
**V version: 0.3.1 fe6197f** 
**OS: ArchLinux** 

**What did you do?**
```v
#include <netdb.h>

struct C.hostent{
	h_name      &char
	h_aliases   &&char
	h_addrtype  int
	h_length    int
	h_addr_list &&char
}

fn main(){
	mut ent := &C.hostent{}
	dump(ent)
}
```

**What did you expect to see?**
No error like
```v
struct C.hostent{
	h_name      &char
	// h_aliases   &&char
	h_addrtype  int
	h_length    int
	// h_addr_list &&char
}

```
 
**What did you see instead?**
```
==================
/tmp/v_1000/bug.1401984773939133309.tmp.c:2069: error: type mismatch in conditional expression (have 'struct string' and 'char *')
...
==================
```

**Comment by spytheman** [comment]
Generated code that triggered the error:
```c
        {_SLIT("\n"), 0xfe10, {.d_s=indents}}, {_SLIT("    h_aliases: &"), 0, {.d_c=0}}, {_SLIT(""), 7, {.d_i32=isnil(it.h_aliases) ? _SLIT("nil") : *it.h_aliases}}, {_SLIT(""), 0, {.d_c=0}},
```

Generated working code for the &char field:
```c
        {_SLIT0, 0xfe10, {.d_s=indents}}, {_SLIT("    h_name: &C"), 0, {.d_c=0}}, {_SLIT("\""), 16, {.d_s=tos4((byteptr)(voidptr) it.h_name)}}, {_SLIT("\""), 0, {.d_c=0}},
```

**Comment by spytheman** [comment]
Should be fixed in 7f2d731 .
